base/random.jl

```diff
@@ -1408,14 +1408,25 @@ end
 Base.show(io::IO, u::UUID) = write(io, Base.repr(u))
 
 # return a random string (often useful for temporary filenames/dirnames)
+
 let b = UInt8['0':'9';'A':'Z';'a':'z']
     global randstring
-    randstring(r::AbstractRNG, n::Int) = String(b[rand(r, 1:length(b), n)])
-    randstring(r::AbstractRNG) = randstring(r,8)
-    randstring(n::Int) = randstring(GLOBAL_RNG, n)
-    randstring() = randstring(GLOBAL_RNG)
+    randstring(r::AbstractRNG, chars::AbstractArray{<:Union{UInt8,Char}}=b, n::Int=8) =
```

Comment:
allowing UInt8 arrays here is a little odd

(and unrelated to this PR, but anyone else think the default length of 8 is kind of arbitrary? was added without much justification in https://github.com/JuliaLang/julia/commit/399c54165093087d52d34cc3c42b1714e795c258)